tools/vm-builder/main.go

```diff
@@ -247,21 +247,41 @@ func main() {
 	}
 	defer cli.Close()
 
-	// pull source image
-	log.Printf("Pull source docker image: %s\n", *srcImage)
-	pull, err := cli.ImagePull(ctx, *srcImage, types.ImagePullOptions{})
+	hostContainsSrcImage := false
+	hostImages, err := cli.ImageList(ctx, types.ImageListOptions{})
 	if err != nil {
 		log.Fatalln(err)
 	}
-	defer pull.Close()
-
-	// do quiet pull - discard output
-	io.Copy(io.Discard, pull)
-	/*
-	   if err = printReader(pull); err != nil {
-	   	log.Fatalln(err)
-	   }
-	*/
+
+	for _, img := range hostImages {
+		for _, name := range img.RepoTags {
+			if name == *srcImage {
+				hostContainsSrcImage = true
+				break
+			}
+		}
+		if hostContainsSrcImage {
+			break
+		}
+	}
+
+	if !hostContainsSrcImage {
+		// pull source image
+		log.Printf("Pull source docker image: %s\n", *srcImage)
```

Comment:
I'm pretty sure that the `\n` is added by default to the end of all log messages if not there already: see https://cs.opensource.google/go/go/+/refs/tags/go1.19.3:src/log/log.go;drc=b45cb0ce863f3cf91a1847d44a9474aad2a6c592;l=171

 The confusing thing here is that this isn't the case with `fmt.Printf` (but is true with `fmt.Println` but the log messages are all new-line terminated.